psi4/driver/procrouting/scf_proc/subclass_methods.py

```diff
@@ -3,18 +3,35 @@
 from psi4 import core
 from ..diis import DIIS, StoragePolicy, RemovalPolicy
 
+def diis_engine_helper(self):
+    engines = set()
+    if core.get_option('SCF', 'DIIS'):
+        engines.add('diis')
+    restricted_open = self.same_a_b_orbs() and not self.same_a_b_dens()
+    if not restricted_open:
+        aediis = core.get_option('SCF', 'INITIAL_SCF_ACCELERATOR')
+        if aediis != "NONE":
+            engines.add(aediis.lower())
+    return engines
+
 def _RHF_orbital_gradient(self, save_fock: bool, max_diis_vectors: int) -> float:
     gradient = self.form_FDSmSDF(self.Fa(), self.Da())
 
     if save_fock:
         if not self.initialized_diis_manager_:
             storage_policy = StoragePolicy.InCore if self.scf_type() == "DIRECT" else StoragePolicy.OnDisk
-            self.diis_manager_ = DIIS(max_diis_vectors, "HF DIIS vector", RemovalPolicy.LargestError, storage_policy)
-            self.diis_manager_.set_error_vector_size(gradient)
-            self.diis_manager_.set_vector_size(self.Fa())
+            self.diis_manager_ = DIIS(max_diis_vectors, "HF DIIS vector", RemovalPolicy.LargestError, storage_policy, engines=diis_engine_helper(self))
             self.initialized_diis_manager_ = True
-        
-        self.diis_manager_.add_entry(gradient, self.Fa())
+
+        entry = {"target": [self.Fa()]}
```

Comment:
Maybe all the `entry` dicts could become a `dataclass`? This would enable tab-completion for the fields `target`/`gradient`/`energy` and avoid misspelling dict strings for these in the long run. Plus, one would have built-in type checks for the fields.

Reply:
I haven't heard of `typeclass` before. How can you make that work when some fields may not be provided? Example: a pure DIIS doesn't need to supply "densities", but ADIIS does.

Reply:
Ok, I see. Such fields can be marked as `Optional` or receive a default value like `densities: List[float] = None` or sth of the sort.

I don't know if it's a really good suggestion I made here, I just thought that using a class would be more pythonic...

I was of course wrong about the built-in type checks, which apparently don't exist 😢 I thought `dataclass` checked whether argument types matched the annotated type, but this was only my wishful thinking...

Reply:
Let's keep the structure as is - if a more complicated `entry` container is needed in the future, one can always refactor.